### ISSUE TITLE
When `ANSI` is enabled, `nu_ansi_term::enable_ansi_support()` should be called on `Windows`

### ISSUE THREAD

**Issue by czy-29** [tokio-rs/tracing#3068]
### Version

tracing-subscriber: v0.3.18

### Platform

64-bit (Windows)

### Crates

tracing-subscriber

### Description

When `ANSI` is enabled, `nu_ansi_term::enable_ansi_support()` should be called on `Windows`, otherwise the ANSI color will appear as garbled (as it is now).

I know users can manually call this to solve the problem, but I believe that `tracing-subscriber` should provide an out-of-the-box solution for this, and the call should be transparent to users. Therefore, I think this is a bug that needs to be fixed.

**Comment by kaffarell** [comment]
I think this is fine, the problem is just where to put it... We can't put it in the (fmt) Subscriber::with_ansi function, because that's not the only place we set the ansi attribute. We would need to add this everywhere.

**Comment by czy-29** [comment]
> I think this is fine, the problem is just where to put it... We can't put it in the (fmt) Subscriber::with_ansi function, because that's not the only place we set the ansi attribute. We would need to add this everywhere.

Yes, when I wrote this issue, I originally intended to submit a PR directly, but later I realized it wasn't that simple... The key issue is, as you said, where should we put it... However, no matter how it is implemented, I believe this logic should indeed be part of the `fmt` implementation.
